src/ros_filter.cpp

```diff
@@ -1945,8 +1945,10 @@ namespace RobotLocalization
             tf2::fromMsg(worldBaseLinkTransMsg_.transform, worldBaseLinkTrans);
 
             tf2::Transform baseLinkOdomTrans;
-            tf2::fromMsg(tfBuffer_.lookupTransform(baseLinkFrameId_, odomFrameId_, ros::Time(0)).transform,
-                         baseLinkOdomTrans);
+            tf2::fromMsg(
+              tfBuffer_.lookupTransform(baseLinkFrameId_, odomFrameId_, filteredPosition.header.stamp, tfTimeout_)
```

Comment:
I actually wonder if I should use `lookupTransformSafe` here, as I want to make sure that we still get _a_ transform if the timeout provided is 0. But I can worry about that upstream.

Reply:
I think that makes sense -- try the correct timestamp, fallback to the current behavior if the tf_timeout elapses. I'll make a PR for upstream today.